lib/graphql/batch.rb

```diff
@@ -15,31 +15,21 @@ def self.batch(executor_class: GraphQL::Batch::Executor)
       end
     end
 
-    def self.use(schema_defn, executor_class: GraphQL::Batch::Executor)
-      # Support 1.10+ which passes the class instead of the definition proxy
-      schema = schema_defn.is_a?(Class) ? schema_defn : schema_defn.target
-      current_gem_version = Gem::Version.new(GraphQL::VERSION)
-      if current_gem_version >= Gem::Version.new("1.6.0")
-        instrumentation = GraphQL::Batch::SetupMultiplex.new(schema, executor_class: executor_class)
-        schema_defn.instrument(:multiplex, instrumentation)
-        if schema.mutation
-          if current_gem_version >= Gem::Version.new('1.9.0.pre3') &&
-              (schema.mutation.is_a?(Class) || schema.mutation.metadata[:type_class])
-            require_relative "batch/mutation_field_extension"
-            schema.mutation.fields.each do |name, f|
-              field = f.respond_to?(:type_class) ? f.type_class : f.metadata[:type_class]
-              field.extension(GraphQL::Batch::MutationFieldExtension)
-            end
-          else
-            schema_defn.instrument(:field, instrumentation)
+    def self.use(schema, executor_class: GraphQL::Batch::Executor)
+      instrumentation = GraphQL::Batch::SetupMultiplex.new(schema, executor_class: executor_class)
+      schema.instrument(:multiplex, instrumentation)
+      if schema.mutation
+        if schema.mutation.is_a?(Class) || schema.mutation.metadata[:type_class]
+          require_relative "batch/mutation_field_extension"
+          schema.mutation.fields.each do |name, f|
+            field = f.respond_to?(:type_class) ? f.type_class : f.metadata[:type_class]
+            field.extension(GraphQL::Batch::MutationFieldExtension)
           end
+        else
+          schema.instrument(:field, instrumentation)
```

Comment:
🤔  there's actually a lot more we can get rid of now by dropping support for < 1.10 (and hence legacy non-class based types).

This conditional would go away entirely and the older `GraphQL::Batch::Setup` would be dropped and merged into `SetupMultiplex`: https://github.com/Shopify/graphql-batch/blob/9fa2dd255e7d0801271471566e3c80083f0adbea/lib/graphql/batch/setup.rb#L2

(`SetupMultiplex` is newer but references methods from `Setup`)

Do you want to take care of those as well in this PR?